src/scheduler/job_info.c

```diff
@@ -5255,6 +5255,16 @@ static int cull_preemptible_jobs(resource_resv *job, void *arg)
 				if (find_node_by_host(job->ninfo_arr, hreq->res_str) != NULL)
 					return 1;
 			} else {
+				if (inp->err->rdef->type.is_non_consumable) {
+				    /* non consumables are used for node selection so
+				     * comparing preemptor and candidate is not sufficient to cull;
+				     * nodes the candidate is using might have the correct tags 
+				     * even if the candidate did not request the non-consumable.
+				     * This will be checked in the node utility test in
+				     * select_index_to_preempt
+				     */
```

Comment:
This comment is a little too wordy for my tastes.  How about we say "In the non-consumable case, we need to pass the job on.  There is a case when a job requesting a non-specific select is allocated a node with a non-consumable.  We will check nodes to see if they are useful in select_index_to_preempt()"

Reply:
Will update.

Reply:
Updated.